abi/abi.go

```diff
@@ -220,8 +224,27 @@ func clone(b []byte) []byte {
 	return result
 }
 
-// QuoteToProto creates a pb.QuoteV4 from the Intel's attestation quote byte array in Intel's ABI format.
-func QuoteToProto(b []uint8) (*pb.QuoteV4, error) {
+// determineQuoteFormat returns the quote format version from the header.
+func determineQuoteFormat(b []uint8) uint32 {
+	data := clone(b)
+	header := &pb.Header{}
+	header.Version = uint32(binary.LittleEndian.Uint16(data[headerVersionStart:headerVersionEnd]))
+	return header.Version
+}
+
```

Comment:
This function should first check size of input array. Incase of an empty array it may give an error while reading version. 
Also, Can you please add a testcase where an empty array is passed to QuoteToProto().

Reply:
Added checks and testcase. Good catch!